controllers/vspheredeploymentzone_controller.go

```diff
@@ -164,13 +164,41 @@ func (r vsphereDeploymentZoneReconciler) Reconcile(ctx goctx.Context, request re
 }
 
 func (r vsphereDeploymentZoneReconciler) reconcileNormal(ctx *context.VSphereDeploymentZoneContext) (reconcile.Result, error) {
-	authSession, err := r.getVCenterSession(ctx)
+	deploymentZoneList := &infrav1.VSphereDeploymentZoneList{}
+	err := r.Client.List(ctx, deploymentZoneList)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	vsphereClusterList := &infrav1.VSphereClusterList{}
+	if err := r.Client.List(ctx, vsphereClusterList, client.InNamespace(r.Namespace)); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if len(vsphereClusterList.Items) != 1 {
+		r.Logger.V(0).Info("more than one vspherecluster in namespace")
+		return reconcile.Result{}, nil
+	}
+
+	allDeploymentZonesReady := true
+	for _, deploymentZone := range deploymentZoneList.Items {
+		if shouldIncludeZone(deploymentZone, &vsphereClusterList.Items[0]) && deploymentZone.Status.Ready != nil && !*deploymentZone.Status.Ready {
```

Comment:
is it possible deploymentZone.Status.Ready is nil? What's should be allDeploymentZonesReady in that case?